globalmangrovewatch/gmw-users
# Feature/jwt auth request

# Description
- Changed auth services to hooks so state can be propagated to other components
- Added axios interceptor which attaches the header `Authorization: Bearer {token}` to MRTT api calls

Fixes # (issue)

## Type of change

Please delete options that are not relevant.


# Instructions

How to test:

1. Check out branch `jwt-auth` of `gmw-api`
2. Follow `gmw-api`'s `README` on how to start the API locally
3. Update `gmw-users`'s 2 related env var (via `.env`):
```
REACT_APP_API_URL=http://localhost:3000/api/v2
REACT_APP_AUTH_URL=http://localhost:3000
```
5. Run `gmw-users` locally
6. Trying to access any routes authenticated should bring to you login page
7. Click `Sign-up` button to sign up
8. Use the account you just created to sign-in

# How Has This Been Tested?

- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration


# Checklist:
